### ISSUE TITLE
Flatpak Support

### ISSUE THREAD

**Issue by JoshuaFern** [satisfactorymodding/SatisfactoryModManager#144]
I play using Steam in a Flatpak. Just running the appimage doesn't detect my install.

#### Here's what I currently have to do to get this working:

* Download the Flatpak, move it into a folder i.e. `~/.var/app/com.valvesoftware.Steam/Downloads`
* Shell into the Flatpak environment using: `flatpak run --command="/bin/bash" com.valvesoftware.Steam`
* Now I just need to `chmod +x`, but wait! I can't just run the appimage because *Flatpaks cannot run AppImages*, so I have to extract it with: `./Satisfactory-Mod-Manager.AppImage --appimage-extract`
* Now I have the loose files in a `squashfs-root/` folder, I could just run `./satisfactory-mod-manager-gui` but there's another problem, CEF is sandboxed by default, so we actually need to run `./satisfactory-mod-manager-gui --no-sandbox` to get the browser to run in our nosuid environment.
* Okay, still with me? Now comes the fun part. We can add our mods! But don't hit that giant launch button at the bottom, it will start another instance of Steam forked from this isolated shell environment. Instead, just start the game from Steam normally.
* From here, we can add the launcher to Steam as a non-steam game. Simply hit the ADD A GAME button on the bottom left of Steam > Add a Non-Steam Game... > Browse to your extracted folder and set the file type filter to all. Then you can add `satisfactory-mod-manager-gui` and right click it > Properties and set --no-sandbox to the Launch Options. You can launch the mod manager easily from Steam now.

#### Proposed solution to avoid all of the above:
Just add the Flatpak search paths to SatisfactoryModManager's game detection code. That way it can manage mods in a flatpak without running in said Flatpak.

**Comment by mircearoata** [comment]
SMM already checks for Steam flatpak installs (https://github.com/mircearoata/SatisfactoryModManagerAPI/blob/master/src/installfinders/linux/steamFlatpak.ts). Do you have flatpak set to install it to somewhere other than `~/.var/app/com.valvesoftware.Steam`? The SMM log (`~/.cache/SatisfactoryModManager/logs`) would help.

**Comment by JoshuaFern** [comment]
Thanks for the quick response. You're right that it does seem to be looking for Flatpak Steam, however it doesn't seem to be finding it for me:

``` ▲ .cache/SatisfactoryModManager/logs cat SatisfactoryModManager-2021-12-26.log
2021-12-26 22:11:23:216	[DEBUG]	Failed to register win-ca
2021-12-26 22:11:23:227	[INFO]	Set debug mode to false
2021-12-26 22:11:24:892	[ERROR]	Command failed: lutris -lj
Trace
Error: Command failed: lutris -lj
    at checkExecSyncError (node:child_process:835:11)
    at Object.execSync (node:child_process:909:15)
    at Object.execSync (node:electron/js2c/asar_bundle:5:12692)
    at Object.e.getInstalls (file:///home/jdf/.cache/appimage-run/d9b7884bc6ee2e733f79a49125a25804c0fd926597759f655e544cceebf80eac/resources/app.asar/dist/electron/renderer.js:301:7301)
    at Object.e.getInstalls (file:///home/jdf/.cache/appimage-run/d9b7884bc6ee2e733f79a49125a25804c0fd926597759f655e544cceebf80eac/resources/app.asar/dist/electron/renderer.js:301:5854)
    at e.getInstalls (file:///home/jdf/.cache/appimage-run/d9b7884bc6ee2e733f79a49125a25804c0fd926597759f655e544cceebf80eac/resources/app.asar/dist/electron/renderer.js:301:5604)
    at file:///home/jdf/.cache/appimage-run/d9b7884bc6ee2e733f79a49125a25804c0fd926597759f655e544cceebf80eac/resources/app.asar/dist/electron/renderer.js:2:306797
    at c (file:///home/jdf/.cache/appimage-run/d9b7884bc6ee2e733f79a49125a25804c0fd926597759f655e544cceebf80eac/resources/app.asar/dist/electron/renderer.js:2:3235899)
    at Generator._invoke (file:///home/jdf/.cache/appimage-run/d9b7884bc6ee2e733f79a49125a25804c0fd926597759f655e544cceebf80eac/resources/app.asar/dist/electron/renderer.js:2:3235687)
    at Generator.next (file:///home/jdf/.cache/appimage-run/d9b7884bc6ee2e733f79a49125a25804c0fd926597759f655e544cceebf80eac/resources/app.asar/dist/electron/renderer.js:2:3236328)
2021-12-26 22:11:24:892	[DEBUG]	Legendary is not installed
2021-12-26 22:11:24:893	[DEBUG]	Steam is not installed
2021-12-26 22:11:24:921	[DEBUG]	Steam-flatpak is not installed
 ```

Steam is certainly installed in the default directory. Clicking the "generate debug info" text in the error message crashes the application:
```
Satisfactory-Mod-Manager.AppImage installed in /home/jdf/.cache/appimage-run/d9b7884bc6ee2e733f79a49125a25804c0fd926597759f655e544cceebf80eac
02:57:41.817 › Checking for update
02:57:42.977 › Update for version 2.7.3 is not available (latest version: 2.7.3, downgrade is disallowed).

(satisfactory-mod-manager-gui:712859): GLib-GIO-ERROR **: 02:57:46.351: No GSettings schemas are installed on the system
[1]    712857 trace trap (core dumped)  appimage-run Satisfactory-Mod-Manager.AppImage

```

This was all run using NixOS `appimage-run` command. NixOS is a unique environment and that might be a factor in the issue we see here.

**Comment by JoshuaFern** [comment]
This has gone stale, closing.
